Cannot create catkin package due to egg_base error

Right now I'm porting the `catkin` package on Arch Linux to Noetic. Catkin 0.8.1 works fine, 0.8.3 not. The `package()` step executes `make DESTDIR=xxx install` which fails:

```
+ cd /home/hermann/ros-noetic-arch/ros-noetic-catkin/src/catkin-0.8.3
+ mkdir -p /home/hermann/ros-noetic-arch/ros-noetic-catkin/pkg/ros-noetic-catkin//opt/ros/noetic/lib/python3.8/site-packages
+ /usr/bin/env PYTHONPATH=/opt/ros/noetic/lib/python3.8/site-packages:/home/hermann/ros-noetic-arch/ros-noetic-catkin/src/build/lib/python3.8/site-packages: CATKIN_BINARY_DIR=/home/hermann/ros-noetic-arch/ros-noetic-catkin/src/build /usr/bin/python3 /home/hermann/ros-noetic-arch/ros-noetic-catkin/src/catkin-0.8.3/setup.py egg_info --egg-base /opt/ros/noetic/lib/python3.8/site-packages build --build-base /home/hermann/ros-noetic-arch/ros-noetic-catkin/src/build install --root=/home/hermann/ros-noetic-arch/ros-noetic-catkin/pkg/ros-noetic-catkin/ --prefix=/opt/ros/noetic --install-scripts=/opt/ros/noetic/bin
running egg_info
error: error in 'egg_base' option: '/opt/ros/noetic/lib/python3.8/site-packages' does not exist or is not a directory
CMake Error at catkin_generated/safe_execute_install.cmake:4 (message):
  
  execute_process(/home/hermann/ros-noetic-arch/ros-noetic-catkin/src/build/catkin_generated/python_distutils_install.sh)
  returned error code
Call Stack (most recent call first):
  cmake_install.cmake:171 (include)


make: *** [Makefile:82: install] Error 1
```

`ros-noetic-catkin` is the first package to install to `/opt/ros/noetic` because all other packages depend on it, so it does not exist yet.

This might be related to https://github.com/ros/catkin/pull/1073.

--- a/cmake/catkin_python_setup.cmake
+++ b/cmake/catkin_python_setup.cmake
@@ -143,7 +143,7 @@ function(catkin_python_setup)
 
   assert(PYTHON_INSTALL_DIR)
   if(${PROJECT_NAME}_SETUP_PY_SETUP_MODULE STREQUAL "setuptools")
-    set(SETUPTOOLS_EGG_INFO "egg_info --egg-base ${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_DIR}")
+    set(SETUPTOOLS_EGG_INFO "egg_info --egg-base ${CMAKE_CURRENT_BINARY_DIR}")
   else()
     set(SETUPTOOLS_EGG_INFO "")
   endif()

